source/cl/source/extension/source/intel_unified_shared_memory/intel_unified_shared_memory.cpp

```diff
@@ -37,10 +37,11 @@ cargo::expected<cl_mem_alloc_flags_intel, cl_int> parseProperties(
       switch (property) {
         case CL_MEM_ALLOC_FLAGS_INTEL: {
           if (0 == (seen & CL_MEM_ALLOC_FLAGS_INTEL)) {
-            if (value & (CL_MEM_ALLOC_INITIAL_PLACEMENT_DEVICE_INTEL |
-                         CL_MEM_ALLOC_INITIAL_PLACEMENT_HOST_INTEL)) {
-              // These options are valid only for shared memory allocations,
-              // which we don't yet support.
+            const auto PLACEMENT = CL_MEM_ALLOC_INITIAL_PLACEMENT_DEVICE_INTEL |
+                                   CL_MEM_ALLOC_INITIAL_PLACEMENT_HOST_INTEL;
+
+            if ((value & PLACEMENT) == PLACEMENT) {
```

Comment:
This style works only if we have exactly two flags. Personally, I think this would be more readable as
```c++
switch (value & PLACEMENT) {
case CL_MEM_ALLOC_INITIAL_PLACEMENT_HOST_INTEL:
  ...; break;
case CL_MEM_ALLOC_INITIAL_PLACEMENT_DEVICE_INTEL:
  ...; break;
default:
  return cargo::make_unexpected(CL_INVALID_PROPERTY);
}
```

Reply:
We only have two "location" flags as per the spec, and I don't see them adding any more ones. With that being the case, I'm not sure the switch statement here is more readable. Especially given that we don't do anything here with the arguments we parse, so they'd just be bare break statements.